src/stateManager/StateManager.ts

```diff
@@ -650,6 +651,129 @@ class StateManager {
         }
     }
 
+    /**
+     * Updates the state snapshot when the fork is different
+     */
+    public async updateSnapshotFork(): Promise<void> {
+        try {
+            // Get the current on-chain snapshot first
+            const currentOnChainSnapshot = StateSnapshot.from(
+                await this.stateChannelManagerContract.getStateSnapshot(
+                    this.channelId
+                )
+            );
+
+            let currentForkId = currentOnChainSnapshot.forkId;
+
+            // Traverse through dispute windows until we reach a fork with no disputes
+            let isDisputed =
+                await this.stateChannelManagerContract.isForkDisputed(
+                    this.channelId,
+                    currentForkId
+                );
+
+            while (isDisputed) {
+                // If reduced result already exists on-chain, traverse to it
+                const existingReducedResult =
+                    await this.stateChannelManagerContract.getReducedResult(
+                        this.channelId,
+                        currentForkId
+                    );
+                if (existingReducedResult[0]) {
```

Comment:
add a comment above `// if reduceResult exists and is final` since the logic for this would change if we we're doing commitments - so we have it in mind later